docs/guides/enable-auto-login.md

```diff
@@ -0,0 +1,53 @@
+# Enabling auto login
+
+If you don't need to handle different users, the auto login setup provided by
+the cypress commands library will add some hooks, using the `cy.login` command,
+that's also provided by the same library, to ensure that each time the correct
+user is being logged in.
+
+## Setting up the auto login
+
+### enabling the auto login
+
+First you need to enable the functionality. This can be done by running the
+appropriate function in the support file (the default is
+`cypress/support/index.js`)
+
+```js
+import { enableAutoLogin } from '@dhis2/cypress-commands'
+
+enableAutoLogin()
+```
+
+### Providing the login data
+
+The setup function uses three environment variables to login:
+
+1. `dhis2_base_url`
+1. `dhis2_username`
+1. `dhis2_password`
+
+There are two places that you can use to put the three required values into:
+
+1. `cypress.json`<br />
+   **This method should only be used if you don't mind the credentials to be
+   versioned by git. This could be a potential security risk.**
+   In the cypress config file, you can place the backend url and the login
+   credentials.<br />
+   The official [cypress
+   docs](https://docs.cypress.io/guides/references/configuration.html) contain
+   all the relevant information on how to add environment variables
+1. `cypress.env.json`<br />
+   This is the safer alternative as this file can be ignored by git. It also
+   allows different developers to use different instance urls without having to
+   change a versioned file
+
+Here is an example what the environment file could look like
+
+```json
+{
+    "dhis2_base_url": "",
+    "dhis2_username": "",
+    "dhis2_password": ""
+}
+```
```

Comment:
I made a comment about this topic earlier, but the additional info I requested for the `cy.login` command is actually here already. Perhaps it would make sense to colocate the info about the env variables with the `cy.login` command and reduce the amount of info you are providing here. You could just explain that `enableAutoLogin` calls the `cy.login` command in the before hook and and makes sure the session is persisted in the `beforeEach` hook?

Reply:
I was thinking about this one a bit more... You've mentioned two ways to set env variables, but there are actually 6 ways to do it. I think we should not include any instructions in this repo, but simply link to https://docs.cypress.io/guides/guides/environment-variables.html

Reply:
> Perhaps it would make sense to colocate the info about the env variables with the cy.login command and reduce the amount of info you are providing here.

It's important to note that this is a guide, not api docs, so I think the duplication is actually good. It should be part of a guide, not putting it into the docs to avoid duplication would be bad as the docs would be incomplete.

I'll add a link to all the possible ways to provide the env vars to the login api docs

> I think we should not include any instructions in this repo

I disagree. The reason why I've put this into the guide is to specifically inform the reader about git-ignoring the `cypress.env.json` so they can have their own backend instance and login credentials, while the data in `cypress.json` could be used for CI

Reply:
> It's important to note that this is a guide, not api docs, so I think the duplication is actually good. It should be part of a guide, not putting it into the docs to avoid duplication would be bad as the docs would be incomplete.

💯 

> I disagree. The reason why I've put this into the guide is to specifically inform the reader about git-ignoring the `cypress.env.json`

Good point, very true. But maybe you could rephrase a bit then: first just explain how to set these vars (offer a link to the cypress docs), and then explain that security might be an issue. I don't think putting variables in the `cypress.json` is the only way to leak secrets. Other possibilities are:
- Some CI config file that defines env vars. To avoid this you would have to use some secrets there
- A dev might define some script in `package.json` that sets an env variable before it runs a command

So maybe some general words of caution would be good, plus perhaps some specific instructions about using a `cypress.env.json` file. Come to think of it, if you are mentioning gitignoring this file, perhaps you should also outline an approach to make sure the correct vars are available in CI too?

Reply:
Included links for setting up github action env vars and encrypted secrets